apps/expo/src/components/Discussion.tsx

```diff
@@ -38,6 +38,50 @@ export default function Discussion({
   const [showCommentInput, setShowCommentInput] = useState(false);
   const [commentText, setCommentText] = useState("");
 
+  const [isLiked, setIsLiked] = useState(false);
+  const [likesCount, setLikesCount] = useState(0);
+
+  const { data: likesData } = api.like.getLikesForDiscussion.useQuery({
+    postId: discussion.id,
+  });
+
+  useEffect(() => {
+    if (likesData) {
+      setIsLiked(likesData.isLikedByUser);
+      setLikesCount(likesData.likesCount);
+    }
+  }, [likesData]);
+
+  const likeMutation = api.like.likePost.useMutation({
+    onSuccess: () => {
+      setIsLiked(true);
+      setLikesCount((prev) => prev + 1);
+    },
+    onError: () => {
```

Comment:
When a like is failed, do you mind explaining why you decide to decrement the count? The `onSuccess` branch is clear, you update the state. Is it that you are updating the count before this mutation so you use the minus to revert?